### DISCUSSION TITLE
Error: unsupported: JOIN not supported between derived tables

### DISCUSSION THREAD

**Discussion by jinhojang6** [planetscale/discussion#465] [Q&A]
I'd like to use planetscale with NestJS and TypeORM but encountered this error.

Error: unsupported: JOIN not supported between derived tables.

Also got this message: QueryFailedError: can't push predicates on concatenate

Anyone solved this issue?

**Comment by orware** [comment]
Hi @jinhojang6,

I'm reviewing things on our end here and I believe a backported solution for this particular issue may have made its way into our environment two days ago, but depending on whether your branch is a production one, you may not see it for another week or two.

If you want me to check on your specific database branch, please reach out to https://support.planetscale.com/ and I can check if your branch is on that latest version that includes the fix already.

**Reply by jinhojang6** [reply]
Thanks for the reply @orware . I submitted a case via https://support.planetscale.com/

**Reply by marcin-czaniecki** [reply]
Hi @jinhojang6,

Do you know what was the reason for the error?

**Reply by jinhojang6** [reply]
Hi, There was an issue with TypeORM. You can ping @orware  for further details.

**Comment by marcin-czaniecki** [comment]
@orware Can you explain what was the problem with TypeORM that @jinhojang6 had? I probably have this problem too. I will be grateful for your help.

**Reply by orware** [reply]
Hi @janossik,

I will point over to this related discussion and reply: https://github.com/planetscale/discussion/discussions/483#discussioncomment-6168463

I would say the wait may almost be over as Vitess 17 was officially released earlier this week, so now just a little more waiting until we have it within PlanetScale's environment and it should hopefully fully address the remaining issue that has been resulting in difficulties for our TypeORM users.
